invenio_rdm_records/services/services.py

```diff
@@ -340,18 +370,19 @@ def reserve_pid(self, id_, identity, pid_type, pid_client=None):
         provider.reserve(pid, draft)
         draft.commit()
 
+        db.session.commit()
```

Comment:
doubt: it's needed because the nested session in the reserve?

Reply:
without this, there is no `db.session.commit` happening anywhere, since it is a specific new endpoint. Or were you thinking to something else?

Reply:
discussed IRL. `draft.commit` adds to the session but does not commit itself....